src/pages/TodosPage.tsx

```diff
@@ -0,0 +1,218 @@
+    if (err !== Error.None) {
+      setTimeout(() => setErrorType(Error.None), 3000);
+    }
+
+    setErrorType(err);
+  };
+
+  useEffect(() => {
+    if (USER_ID) {
+      getTodos(USER_ID)
+        .then(result => {
+          setTodos(result);
+        })
+        .catch(() => handleError(Error.Load));
+    }
+  }, [USER_ID]);
+
+  const todosStatus = useMemo(() => {
+    return {
+      isActive: todos.some(todo => todo.completed === false),
+      isCompleted: todos.some(todo => todo.completed === true),
+    };
+  }, [todos]);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setTodoCondition(TodoCondition.deleting);
+    setProcesingTodosId([todoId]);
+    deleteTodo(todoId)
+      .then(() => setTodos(prev => prev.filter(({ id }) => id !== todoId)))
+      .catch(() => handleError(Error.Delete))
+      .finally(() => setTodoCondition(TodoCondition.neutral));
+  };
+
+  const clearCompleted = () => {
+    setTodoCondition(TodoCondition.deleting);
+
+    todos?.forEach(todo => {
+      if (todo.completed) {
+        setProcesingTodosId((state) => [...state, todo.id]);
+        deleteTodo(todo.id)
+          .then(() => setTodos(prev => prev.filter(({ id }) => id !== todo.id)))
+          .catch(() => handleError(Error.Delete))
+          .finally(() => setTodoCondition(TodoCondition.neutral));
+      }
+    });
+  };
+
+  const toggleTodos = (curentTodos: Todo[], status?: boolean | undefined) => {
+    setTodoCondition(TodoCondition.saving);
+
+    curentTodos.forEach(curentTodo => {
+      if (curentTodo.completed !== status) {
+        setProcesingTodosId((state) => [...state, curentTodo.id]);
+      }
+
+      const copyTodos = [...todos];
+      const indexCurTodo = copyTodos
+        .findIndex(({ id }) => id === curentTodo.id);
+      const newStatus: boolean = status || !curentTodo.completed;
+
+      copyTodos[indexCurTodo].completed = newStatus;
+
+      changeTodo(curentTodo.id, { completed: newStatus })
+        .then(() => {
+          setTodos(copyTodos);
+        })
+        .catch(() => handleError(Error.Update))
+        .finally(() => {
+          setProcesingTodosId([]);
+          setTodoCondition(TodoCondition.neutral);
+        });
+    });
+  };
+
+  const toggleAllTodos = () => {
+    toggleTodos(todos, todosStatus.isActive);
+  };
+
+  const handleSubmitEditing = (todoId: number, newTitle: string) => {
+    setProcesingTodosId([todoId]);
+    setTodoCondition(TodoCondition.saving);
+
+    changeTodo(todoId, { title: newTitle })
+      .then(() => setTodos(prevTodos => prevTodos.map(todo => {
+        if (todo.id === todoId) {
+          return {
+            ...todo,
+            title: newTitle,
+          };
+        }
+
+        return todo;
+      })))
+      .catch(() => Error.Update)
+      .finally(() => {
+        setProcesingTodosId([]);
+        setTodoCondition(TodoCondition.neutral);
+      });
+  };
+
+  const onLogout = () => {
+    localStorage.removeItem('user');
+    navigate(0);
```

Comment:
Replace 'navigate(0)' with 'navigate('/')' for better clarity.

Reply:
navigate('/') does not reload the page